pkg/controller/buildrun/buildrun_controller.go

```diff
@@ -209,18 +209,23 @@ func (r *ReconcileBuildRun) retrieveServiceAccount(buildRun *buildv1alpha1.Build
 	var serviceAccountName string
 	if buildRun.Spec.ServiceAccount == nil {
 		serviceAccountName = pipelineServiceAccountName
+		err := r.client.Get(context.TODO(), types.NamespacedName{Name: serviceAccountName, Namespace: buildRun.Namespace}, buildServiceAccount)
+		if err != nil && !errors.IsNotFound(err) {
+			log.Error(err, "Failed to get ServiceAccount", "ServiceAccount", serviceAccountName)
+			return nil, err
+		} else if errors.IsNotFound(err) {
+			serviceAccountName = defaultServiceAccountName
+			err = r.client.Get(context.TODO(), types.NamespacedName{Name: serviceAccountName, Namespace: buildRun.Namespace}, buildServiceAccount)
+			if err != nil {
+				log.Error(err, "Failed to get default ServiceAccount")
+				return nil, err
+			}
+		}
 	} else {
 		serviceAccountName = *buildRun.Spec.ServiceAccount
-	}
-	err := r.client.Get(context.TODO(), types.NamespacedName{Name: serviceAccountName, Namespace: buildRun.Namespace}, buildServiceAccount)
-	if err != nil && !errors.IsNotFound(err) {
-		log.Error(err, "Failed to get ServiceAccount", "ServiceAccount", serviceAccountName)
-		return nil, err
-	} else if errors.IsNotFound(err) {
-		serviceAccountName = defaultServiceAccountName
-		err = r.client.Get(context.TODO(), types.NamespacedName{Name: serviceAccountName, Namespace: buildRun.Namespace}, buildServiceAccount)
-		if err != nil {
-			log.Error(err, "Failed to get default ServiceAccount")
+		err := r.client.Get(context.TODO(), types.NamespacedName{Name: serviceAccountName, Namespace: buildRun.Namespace}, buildServiceAccount)
+		if err != nil && !errors.IsNotFound(err) {
+			log.Error(err, "Failed to get ServiceAccount", "ServiceAccount", serviceAccountName)
```

Comment:
Would this get written to the BuildRun status ?

Reply:
No, we didn't write all failure to BuildRun status.

That is also I asked at the meeting. I think maybe from next week, we need to write **ALL** related failture message into BuildRun status, so that end user can know what is wrong

Reply:
Sorry for holding this up, if we aren't writing to the `status`, we will leave the user confused as to why the build never really ran. :(

 Would you be updating this PR to write to the `status` as well ?

Reply:
Agree. And yes, we can use this case as an example for other failure handle. :)

Because there are some overlappings, I merge this PR to the PR: https://github.com/redhat-developer/build/pull/105

And update the status in that PR.

After that PR is merged, we can close this one safely.

Reply:
Sure, thanks!